app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt

```diff
@@ -35,26 +37,29 @@ class SearchLocationFragment : Fragment() {
             thisActivity.supportActionBar?.title = ""
             thisActivity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
         }
-        rootView.locationProgressBar.visibility = View.GONE
+        setHasOptionsMenu(true)
 
         fromSearchFragment = arguments?.getBoolean(FROM_SEARCH) ?: false
 
-        if (thisActivity is Activity) geoLocationUI.configure(thisActivity, rootView, searchLocationViewModel)
+        geoLocationViewModel.getVisibility()
+        geoLocationViewModel.currentLocationVisibility.observe(this, Observer {
+            currentLocation.visibility = View.GONE
+        })
+
+        rootView.currentLocation.setOnClickListener {
+            checkLocationPermission()
+            if (thisActivity is Activity) geoLocationViewModel.configure(thisActivity)
```

Comment:
When will thisActivity not be an activity?

Reply:
thisActivity is activity inside on create view.

Reply:
Why is the if statement required

Reply:
But activity is from getActivity() of type FragmentActivity, need to cast for Activity

Reply:
> When will thisActivity not be an activity?

Okay got it. It is Activity for following if statement: 
```java
val thisActivity = activity
        if (thisActivity is AppCompatActivity) {
            thisActivity.supportActionBar?.title = ""
            thisActivity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
        }
```

Reply:
It doesn't need to be smart casted, just pass it directly

Reply:
It is showing required: Activity found: FragmentActivity

Reply:
FragmentActivity is child of Activity, so there shouldn't be a problem

Reply:
Make it nullable in the receiving function and just return if it is null